### PR TITLE
Maxwell rawio implementation

### PR DESCRIPTION
**Pull request by samuelgarcia**
@yger @alejoe91 
Could you test this ?

### CODE REVIEW

**Comment by samuelgarcia** [comment]
@yger @alejoe91 : can you make feedback on this soon so we could merge this ?

**Comment by samuelgarcia** [comment]
As soon as files are merge in gin. This is ready to review.

**Comment by alejoe91** [comment]
@samuelgarcia maybe a possible solution is to add the libcompression.so in the maxwell folder and add the env variable before testing?

**Comment by samuelgarcia** [comment]
I prefer not include the **libcompression.so** in the code. We need permission for that.
Maybe we could write a **libcompression.so** retriever but I don't have time for that at the moment.
I think the error message is enough and is already a big improvement for user compare to previous extractor no ?

**Comment by alejoe91** [comment]
Yes. I asked maxwell to make it downloadable via wget

**Comment by samuelgarcia** [comment]
OK we can try a:
 * wget in the yml of github actions
 * wget in python code (more work)

**Comment by samuelgarcia** [comment]
@JuliaSprenger : can you have a look to this I patch the download_dataset() and add the update stage.
So distant changes will be updated locally now.
When this will be merge other PR will be OK.

**Comment by samuelgarcia** [comment]
@alejoe91 : I add a util function auto download this plugin.
test are still failing. I prefer to disable for now.
We will see this later.

**Comment by samuelgarcia** [comment]
Ready to review.

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:67]: what should happen in case of `int(version) < 20160704` ? Are you sure no older versions exist?

Comment (alejoe91) [reply neo/rawio/maxwellrawio.py:67]: yes this is the first released version :)

Comment (alejoe91) [reply neo/rawio/maxwellrawio.py:67]: @JuliaSprenger maybe we can add an `else: raise Exception("Unsupported version")` for completeness?

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:78]: Do you mean this?
```suggestion
                        raise ValueError("Detected multiple recordings. Please select a single recording using the `rec_name` paramter."
```

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:90]: same here: are you sure no files exist with older versions?

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:187]:
```suggestion
The internal compression requires a custom plugin!!!
```

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:190]:
```suggestion
Please visit this page and install the missing decompression libraries:
```

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:192]:
```suggestion
Then, link the decompression library by setting the `HDF5_PLUGIN_PATH` to your installation location, e.g. via
```

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:193]: are you sure the user is supposed to overwrite this variable if it already exists? Or should this rather be a list and the new entry needs to be appended?

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:196]:
```suggestion
function that do it automagically.
```

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [inline neo/rawio/maxwellrawio.py:195]:
```suggestion
Alternatively, you can use the auto_install_maxwell_hdf5_compression_plugin() below
```

File: neo/test/rawiotest/test_maxwellrawio.py
Comment (JuliaSprenger) [inline neo/test/rawiotest/test_maxwellrawio.py:11]: Does this mean that the test might fail locally even when the user did install all official requirements? Shouldn't this test rather be only executed if hdf5 is present in the first place?

**Review by JuliaSprenger — APPROVED** [review]

File: neo/rawio/maxwellrawio.py
Comment (samuelgarcia) [reply neo/rawio/maxwellrawio.py:193]: I  most cases this ENV do not exists.
So 99% users, this will help.
For hdf5-advanced users, they will anderstand what to do.

File: neo/test/rawiotest/test_maxwellrawio.py
Comment (samuelgarcia) [reply neo/test/rawiotest/test_maxwellrawio.py:11]: Maybe we should remove this from test always.
The test is working on my local machine.

Comment (JuliaSprenger) [reply neo/test/rawiotest/test_maxwellrawio.py:11]: Ah, earlier I didn't realize you are talking about the compression library here. I think it would be best to check if the plugin is installed and only run the test if this is the case.

So the auto installation does not work on github actions?

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [reply neo/rawio/maxwellrawio.py:193]: ok, let's hope they raise an issue when they encounter problems related to this...

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [reply neo/rawio/maxwellrawio.py:67]: yes, sounds good, just to be on the save side.

File: neo/test/rawiotest/test_maxwellrawio.py
Comment (samuelgarcia) [reply neo/test/rawiotest/test_maxwellrawio.py:11]: No it was working, this is why I disable.
I need to check again.

**Comment by samuelgarcia** [comment]
@apdavison : If this pass tests, could we merge this soon ?
This PR will unblock other PR and will fix the file caching.
Now any PR take more than 1 hour.

**Comment by samuelgarcia** [comment]
Ready to review (and merge maybe)

File: neo/test/rawiotest/test_maxwellrawio.py
Comment (JuliaSprenger) [reply neo/test/rawiotest/test_maxwellrawio.py]: @samuelgarcia did you check again?

Comment (samuelgarcia) [reply neo/test/rawiotest/test_maxwellrawio.py]: This line is outdated. I remove the skip and now the maxwell is tested and test are passing.

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [reply neo/rawio/maxwellrawio.py:67]: @samuelgarcia wdyt?

File: neo/rawio/maxwellrawio.py
Comment (JuliaSprenger) [reply neo/rawio/maxwellrawio.py:90]: @samuelgarcia wdyt?

Comment (samuelgarcia) [reply neo/rawio/maxwellrawio.py:92]: I have add a raise NotImplemented error

**Pull request merged**
